### PR TITLE
Completed solution for work-with-file task

### PR DESCRIPTION
**Pull request by AntonU91**

### CODE REVIEW

File: src/main/java/core/basesyntax/WorkWithFile.java [hunk 1]
```diff
@@ -1,7 +1,43 @@
...
+        }
+        int result = supplyAmount - buyAmount;
+        return String.format("supply,%d%nbuy,%d%nresult,%d", supplyAmount, buyAmount, result);
+    }
 
+    private void writeStatisticToFile(String statistic, String outputFileName) {
+        try (BufferedWriter bfWriter = new BufferedWriter(new FileWriter(outputFileName))) {
+            bfWriter.write(statistic);
+        } catch (IOException e) {
+            e.printStackTrace();
```
Comment (okuzan) [inline src/main/java/core/basesyntax/WorkWithFile.java:40]: image

**Review by okuzan — CHANGES_REQUESTED** [review]
Before you create PR with your homework, you need to go through the checklist under the task and correct all the points described there. The mentor will not check the work until the checklist points are corrected

File: src/main/java/core/basesyntax/WorkWithFile.java [hunk 2]
```diff
@@ -1,7 +1,43 @@
...
+            String str = bfReader.readLine();
+            while (str != null) {
+                if (str.startsWith("b")) {
+                    buyAmount += Integer.parseInt(str.replaceFirst("\\w+,", ""));
+                } else {
+                    supplyAmount += Integer.parseInt(str.replaceFirst("\\w+,", ""));
+                }
+                str = bfReader.readLine();
+            }
+        } catch (IOException e) {
```
Comment (olekskov) [inline src/main/java/core/basesyntax/WorkWithFile.java:29]: Performing business logic during file reading make connection to file to be opened for longer period of time what is not desirable.
Please create separate method to read file content and then proceed with calculations in separate method

**Review by olekskov — CHANGES_REQUESTED** [review]

File: src/main/java/core/basesyntax/WorkWithFile.java [hunk 3]
```diff
@@ -1,7 +1,55 @@
...
+        String results = getStatisticValuesFromRetrievedData(dataToAnalyze, fromFileName);
+        writeStatisticToFile(results, toFileName);
+    }
+
+    private String getStatisticValuesFromRetrievedData(List<String> data, String inputFileName) {
+        int buyAmount = 0;
+        int supplyAmount = 0;
+
+        for (String str : data) {
+            if (str.startsWith("b")) {
```
Comment (aaananas) [inline src/main/java/core/basesyntax/WorkWithFile.java:24]:
```suggestion
            if (str.startsWith("buy")) {
```

**Review by aaananas — APPROVED** [review]

**Pull request closed without merging**